src/snowflake/snowpark/session.py

```diff
@@ -1485,6 +1487,16 @@ def range(self, start: int, end: Optional[int] = None, step: int = 1) -> DataFra
         set_api_call_source(df, "Session.range")
         return df
 
+    @experimental(version="0.12.0")
+    def create_async_job(self, query_id: str) -> AsyncJob:
```

Comment:
How do I know the result type of this query if there is no result type?

Reply:
It seems like `result_type` is part of `AsyncJob#result` call, so user has the flexibility to choose the result type when they need the result. Of course Jianzhun knows better :)

Reply:
yep, and if result_type is not provided, the default value is a list of rows (see constructor of AsyncJob and I mentioned it in the docstring).